File: src/main/java/kz/smarthealth/patientservice/util/MessageSource.java
package kz.smarthealth.patientservice.util;

public enum MessageSource {

    PATIENT_BY_ID_NOT_FOUND("Patient with id=%s not found.");

    private final String text;

    MessageSource(String text) {
        this.text = text;
    }

    public String getText(String... params) {
        return String.format(this.text, params);
    }
}

File: src/main/kotlin/kz/samat/patientservice/enum/MessageSource.kt
package kz.samat.patientservice.enum

enum class MessageSource (val text: String) {

    PATIENT_NOT_FOUND("Patient with such id not found, id: %s"),
    FORBIDDEN("You do not have access to commit this operation");

    fun getText(vararg values: String) = String.format(this.text, *values)
}